server/cluster/cluster.go

```diff
@@ -1114,6 +1114,37 @@ func (c *RaftCluster) buryStore(storeID uint64) error {
 	return err
 }
 
+// ForceRemoveStore set a store as tombstone directly without waiting for the data to be moved somewhere else.
+func (c *RaftCluster) ForceRemoveStore(storeID uint64) error {
```

Comment:
Can we use `BuryStore`?

Reply:
I intentionally created a new function, because, given the fact that [offline](https://github.com/tikv/pd/blob/3448e55fc5e7974e261a282f66204379e70a7855/server/core/store.go#L150) does not mean [disconnected](https://github.com/tikv/pd/blob/3448e55fc5e7974e261a282f66204379e70a7855/server/core/store.go#L475), bury a store that is "up" is not allowed by `buryStore`, but is allowed in unsafe recovery.

Reply:
You can take a look at the implementation of `BuryStore` in release 4.0. We can add a flag to control the behavior instead of creating a new function.

Reply:
BTW, what if we only have a small number of regions that need to be recovered? e.g. we can use `-r` in tikv-ctl.

Reply:
For this kind of operation, we must give a prompt to the users to let the users confirm their operations.

Reply:
> BTW, what if we only have a small number of regions that need to be recovered? e.g. we can use `-r` in tikv-ctl.

@v01dstar Seems, we should support it later

Reply:
> BTW, what if we only have a small number of regions that need to be recovered? e.g. we can use `-r` in tikv-ctl.

Not sure what concerns you in this case, could you be more specific?

I have updated the implementation which reuses the `BuryStore` function, PTAL

Reply:
> Not sure what concerns you in this case, could you be more specific?

In this case, we don't need delete store.

Reply:
It is still better to remove/bury/delete store even if a small number of regions are affected, right? e.g. in "ONCALL-3811", several regions were not able to elect leaders since replicas in the same group had conflicting states, which we believe was due to the restarting of failed stores. This PR prevent such things from happening by marking failed stores as tombstones. Not deleting the store means the store can still be booted? In that case, unsafe recovery should not be used at the first place?

Reply:
What if there are some bugs which lead to some regions unavailable? There is no need to delete the store.

Reply:
Was there an existed issue that required using unsafe recovery to mitigate while having a store to be kept alive? I thought we only use this to `remove failed stores` right?

Reply:
For example: ONCALL-1659

Reply:
I see. Currently, Online unsafe recovery does not support removing failed stores from a specific region (or `-r` option). If we are going to implement that in the future we will make sure we don't mark the store as tombstone, but so far I don't think that's a problem, admin or oncall should use online unsafe recovery if the entire store is broken, and still use tikv ctl unsafe recovery if they want to remove a peer from a specific group.